### DISCUSSION TITLE
ink合约Balance除100或乘小数

### DISCUSSION THREAD

**Discussion by comien** [ParityAsia/substrate-newsletter#13] [Substrate/Polkadot 技术问题]
```
 #[ink(message)]
pub fn percent(&self) -> Balance {
    let total = self.total_supply();
    total / 100
}
```
上传合约报错：system.ExtrinsicFailed Other
🙏请问下，怎么才可以取1%呢，怎么可以乘小数？

**Comment by kaichaosun** [comment]
内置类型有一个 [Percent](https://crates.parity.io/sp_runtime/struct.Percent.html)。

**Reply by comien** [reply]
请问在ink中使用Percent需要dependencies依赖sp_runtime::Percent吗？有没有ink使用Percent的例子呢？

**Reply by kaichaosun** [reply]
我没找到直接使用的例子，这个例子是另外一种解决方案
https://github.com/PolymathNetwork/Polymesh/blob/bae0b8cc4886ba70e3cbe07fd73de9a8c1ba5dd1/contracts/PercentageTransferManager/src/lib.rs#L143
